### PR TITLE
wholebodydynamics: Remove support for reading FT and IMU using IAnalogSensor and IGenericSensor interfaces

### PR DESCRIPTION
**Pull request by traversaro**
Fix https://github.com/robotology/whole-body-estimators/issues/179 .
Fix https://github.com/robotology/whole-body-estimators/issues/170 .

This support was not explicitly declared deprecated in wholebodydynamics, however the related interfaces have been deprecated in YARP for a long time. For users interested in using them, please use whole-body-estimators v0.10.0 . The `HW_USE_MAS_IMU` group is now compulsory.

Further cleanup is probably possible (especially in the FT handling logic), but I did not want to mix up the removal of support from the further cleanup.

### CODE REVIEW

File: devices/wholeBodyDynamics/WholeBodyDynamicsDevice.cpp [hunk 1]
```diff
@@ -1117,7 +1117,7 @@ bool WholeBodyDynamicsDevice::loadSettingsFromConfig(os::Searchable& config)
 
     if( !prop.check("HW_USE_MAS_IMU") )
     {
-        useMasIMU = false;
+        yError() << "wholeBodyDynamics : missing required group parameter HW_USE_MAS_IMU";
```
Comment (S-Dafarra) [inline devices/wholeBodyDynamics/WholeBodyDynamicsDevice.cpp:1120]: Should it ``return false`` as well?

Comment (traversaro) [reply devices/wholeBodyDynamics/WholeBodyDynamicsDevice.cpp:1120]: Yes thanks, done in https://github.com/robotology/whole-body-estimators/pull/181/commits/3d5ff94d8cd63759c7d9657a974fec1f0495db26 .

**Review by S-Dafarra — APPROVED** [review]

**Review by CarlottaSartore — APPROVED** [review]

File: devices/wholeBodyDynamics/WholeBodyDynamicsDevice.cpp [hunk 2]
```diff
@@ -2142,28 +2088,6 @@ bool WholeBodyDynamicsDevice::readIMUSensors(bool verbose)
...
-            rawIMUMeasurements.angularVel(1) = deg2rad(imuMeasurement[7]);
-            rawIMUMeasurements.angularVel(2) = deg2rad(imuMeasurement[8]);
-
-            rawIMUMeasurements.linProperAcc(0) = imuMeasurement[3];
-            rawIMUMeasurements.linProperAcc(1) = imuMeasurement[4];
-            rawIMUMeasurements.linProperAcc(2) = imuMeasurement[5];
-        }
-    }
-    else
     {
```
[the remark this answers is not in the document]
Comment (traversaro) [reply devices/wholeBodyDynamics/WholeBodyDynamicsDevice.cpp:2091]: Yes, I know, but I did not want to add unnecessary noise in the PR, with all those lines that just changed due to indentation.

File: devices/wholeBodyDynamics/WholeBodyDynamicsDevice.h [hunk 3]
```diff
@@ -316,21 +312,19 @@ class WholeBodyDynamicsDevice :  public yarp::dev::DeviceDriver,
     /**
      * Attach all Six Axis Force/Torque devices.
      * A device is identified as a Six Axis F/T if it
-     * implements the IAnalogSensor interface. It also ataches MAS Six Axis F/T sensors.
-     * A device is identified as a MAS Six Axis F/T if it
-     * implements the Multiple Analog Sensor interfaces.
+     * implements the ISixAxisForceTorqueSensors interface
+     * and getNrOfSixAxisForceTorqueSensors() >= 1 .
+     * If the device also implements the ITemperatureSensors
+     * interface, that is handled.
```
[the remark this answers is not in the document]
Comment (traversaro) [reply devices/wholeBodyDynamics/WholeBodyDynamicsDevice.h:318]: Not sure about this comment, what do you mean? That the phrase is not clear?

[the remark this answers is not in the document]
Comment (CarlottaSartore) [reply devices/wholeBodyDynamics/WholeBodyDynamicsDevice.h:318]: Sorry @traversaro   I missed this message !I was not super sure about the meaning of the sentence but it is fine :)

[the remark this answers is not in the document]
Comment (traversaro) [reply devices/wholeBodyDynamics/WholeBodyDynamicsDevice.h:318]: Ack, I will change it in a future PR then. I meant. If the device also implements the `yarp::dev::ITemperatureSensors` interface, this interface will be viewed and configured in this method.

**Pull request merged**